src/nodeViews/createReactNodeViewConstructor.tsx

```diff
@@ -207,11 +241,28 @@ export function createReactNodeViewConstructor(
       />
     );
 
-    const unregisterElement = registerElement(
-      element,
-      dom as HTMLElement,
-      Math.floor(Math.random() * 0xffffff).toString(16)
-    );
+    if (contentDOM) {
+      // Store this node view's registration key on the contentDOM
+      // so it can be retrieved by its children.
+      contentDOM.dataset[REACT_PM_KEY] = key;
+    }
+
+    let unregisterElement: UnregisterElement;
+
+    // ProseMirror hasn't assigned finished constructing the
+    // node view descriptor tree yet, so attempts to ascend it
+    // will fail until after the current call stack has finished
+    // executing.
+    queueMicrotask(() => {
+      batch(() => {
+        unregisterElement = registerElement(
+          findNearestRegistryKey(editorView, getPos()),
+          element,
+          dom as HTMLElement,
+          key
+        );
+      });
+    });
```

Comment:
@tilgovi I feel like you were describing something more clever than this, but... this _will_ work, right? We'll queue one microtask per node view constructor, and they'll all run after the current task queue? Are there performance implications for queuing one microtask per node view constructor rather than one microtask for all of them?

Reply:
> We'll queue one microtask per node view constructor, and they'll all run after the current task queue?

Yes.

> Are there performance implications for queuing one microtask per node view constructor rather than one microtask for all of them?

Yes, both in function overhead and because we're calling `batch` around individual registrations rather than a group of them. I was kind hoping we could batch all of the outstanding registrations.

Reply:
Is this getting too complicated? I'm nervous. 😆

Reply:
Ooooh, wait, if we're not already can't we move this batch call to be around the call to update the editor view and then not worry about this here?

Reply:
Orrr..... are we already and is that _why_ `setState` doesn't actually run its function immediately? 🤯